resources/views/pages-legacy/linkedhashes.blade.php

```diff
@@ -11,39 +12,55 @@
     abort(404);
 }
 
-$gameData = getGameData($gameID);
-if (!$gameData) {
+$game = Game::with('system')->find($gameID);
+if (!$game) {
     abort(404);
 }
 
-$consoleName = $gameData['ConsoleName'];
-$consoleID = $gameData['ConsoleID'];
-$gameTitle = $gameData['Title'];
-$gameIcon = $gameData['ImageIcon'];
-$forumTopicID = $gameData['ForumTopicID'];
-$hashes = getHashListByGameID($gameID);
-?>
-<x-app-layout pageTitle="Supported Game Files - {{ $gameTitle }}">
-    <div class='navpath'>
-        <?= renderGameBreadcrumb($gameData) ?>
-        &raquo; <b>Supported Game Files</b>
-    </div>
+$hashes = $game->hashes()->with('user')->orderBy('name')->orderBy('md5')->get();
 
-    <h3>List of Supported Game Files</h3>
+$numHashes = $hashes->count();
 
-    <?php
-    $numHashes = count($hashes);
+$unlabeledHashes = $hashes->filter(function ($hash) {
+    return empty($hash->name);
+});
+$labeledHashes = $hashes->reject(function ($hash) {
+    return empty($hash->name);
+});
+?>
 
-    echo gameAvatar($gameData, iconSize: 64);
-    echo "<br><br>";
+<x-app-layout pageTitle="Supported Game Files - {{ $game->title }}">
```

Comment:
You're already like 80% of the way to full conversion of this page. Suggest replacing the cookie authentication calls above, the devbox below, and moving this out of pages-legacy with a proper route: "game/[game]/hashes"

Reply:
Good point. Done in latest.

I also took a pass at significantly refactoring the PHP logic in `<x-game.link-buttons.game-link-button />` to:
* Accept a `Game` model as a prop.
* Lean more into a reliance on policies.